src/java.desktop/share/classes/javax/imageio/stream/FileCacheImageOutputStream.java

```diff
@@ -217,6 +229,32 @@ public boolean isCachedMemory() {
         return false;
     }
 
+    private static class FileCacheDisposerRecord implements DisposerRecord {
+
+        private final File cacheFile;
+        private final RandomAccessFile cache;
+        private volatile boolean disposed;
+
+        public FileCacheDisposerRecord(File cacheFile, RandomAccessFile cache) {
+            this.cacheFile = cacheFile;
+            this.cache = cache;
+        }
+
+        @Override
+        public synchronized void dispose() {
+            if (disposed) {
+                return;
+            }
+            try {
+                cache.close();
```

Comment:
I notice a similar nested class/method in FileCacheImageInputStream with a few differences. Does the cache.close() and cacheFile.delete need to be wrapped in null checks? And once deleted do the StreamDisposerRecord pointers (this.cacheFile and this.cache) need to be set to null like in the ImageInputStream version of the code?

Reply:
Use of `final` is recommended wherever possible, as the values are then guaranteed to be visible on other threads (e.g. the disposer thread).

However IIUC, values set in a `DisposerRecord` by a program thread should (already) be visible on the disposer thread because `Disposer.add()` and `Disposer.run()` (via `records.put()` and `records.remove()`, respectively) lock the same `records` `Hashtable` object.